form-builder/index.html

```diff
@@ -0,0 +1,1201 @@
+                            section.subtitle(sectionData.subtitle);
+                            section.repeatable(sectionData.repeatable);
+                            section.newPage(sectionData.newPage);
+                            section.sectionHelpText(sectionData.sectionHelpText);
+                            section.reviewPage(sectionData.reviewPage);
+
+                            if (section.choiceHeaders) {
+                                section.questionHeader(sectionData.questionHeader);
+                                section.choiceHeaders(sectionData.choiceHeaders.map(choiceHeader => ko.observable(choiceHeader)));
+                                section.choiceHeaderDisplayText(sectionData.choiceHeaderDisplayText);
+                                section.tabular(true);
+                            }
+
+                            sectionData.questions.forEach(questionData => {
+
+                                const question = new QuestionViewModel(index + 1, questionData.questionId);
+                                question.title(questionData.title);
+                                question.subtitle(questionData.subtitle);
+                                question.questionHelpText(questionData.questionHelpText);
+                                question.type(questionData.type);
+                                question.choices(questionData.choices?.map(choice => ko.observable(choice)));
+                                question.minimum(questionData.minimum || questionData.exclusiveMinimum);
+                                question.maximum(questionData.maximum || questionData.exclusiveMaximum);
+                                question.exclusiveMinimum(questionData.exclusiveMinimum);
+                                question.exclusiveMaximum(questionData.exclusiveMaximum);
+                                question.multiChoice(questionData.multiChoice);
+                                question.multiLineText(questionData.multiLineText);
+                                question.required(questionData.required);
+                                question.captureTime(questionData.captureTime);
+                                question.captureLocation(questionData.captureLocation);
+                                section.questions.push(question);
+                            });
+
+                            // if questionData is acknowledgement or geolocation, read the 'for' attribute, 
+                            // find the question and set captureTime or captureLocation to true for that question
+                            // and remove the annotated question from the list. 
+                            sectionData.questions.forEach(questionData => {
+
+                                if (questionData.type === 'acknowledgement' || questionData.type === 'geolocation') {
+                                    const question = section.questions().find(q => q.questionId() === questionData.for);
```

Comment:
The `required` property of the capture time and location should be based on the main question.
- If the main question required, then the associated question will be required as well
- otherwise, it will be not required.

Also the ordering of the generated JSON for associated questions should be below the main question that generate from